onnxruntime/core/framework/tunable.h

```diff
@@ -168,7 +171,13 @@ class TunableOp {
     // Do nothing if we are not playing around with params
   }
 
-  virtual ~TunableOp() = default;
+  std::string Signature() {
+    // NOTE: this is not thread-safe, but it seems to be harmless
```

Comment:
I don't think this is as benign as it appears: we can't really assume that string assignment can be accomplished with a single instruction. If the assignment requires more than one instruction, then two assignments from two different threads can interleave, potentially leaving the string object in an invalid state. Such a bug would be extremely hard to reproduce and debug.

Reply:
mutex guarded